sopel_iplookup/plugin.py

```diff
@@ -87,30 +95,58 @@ def _find_geoip_db(bot: SopelWrapper):
                 LOGGER.warning(
                     'GeoIP path configured but DB not found in %s', str(pth))
 
-    LOGGER.info('Downloading GeoIP database')
+    LOGGER.info('Downloading GeoIP database files')
     bot.say('Downloading GeoIP database, please wait...')
 
-    common_params = {
-        'license_key': config.ip.maxmind_license_key,
-        'suffix': 'tar.gz',
-    }
-    base_url = 'https://download.maxmind.com/app/geoip_download'
     geolite_urls = []
-
-    for edition in ['ASN', 'City']:
-        geolite_urls.append(
-            '{base}?{params}'.format(
-                base=base_url,
-                params=web.urlencode(dict(common_params, **{'edition_id': 'GeoLite2-%s' % edition})),
+    editions = ['ASN', 'City']
+
+    if config.ip.maxmind_license_key:
+        # Direct downloads via MaxMind account
+        base_url = 'https://download.maxmind.com/app/geoip_download'
+        common_params = {
+            'license_key': config.ip.maxmind_license_key,
+            'suffix': 'tar.gz',
+        }
+
+        for edition in editions:
+            geolite_urls.append(
+                '{base}?{params}'.format(
+                    base=base_url,
+                    params=web.urlencode(
+                        dict(
+                            common_params,
+                            **{'edition_id': 'GeoLite2-%s' % edition}
+                        )
+                    ),
+                )
             )
+    else:
+        # Indirect downloads with no key needed
+        base_url = (
+            'https://raw.githubusercontent.com'
+            '/P3TERX/GeoLite.mmdb/download/GeoLite2-{edition}.mmdb'
```

Comment:
This looks like we should have a separate `_download_geoip()` helper (or two) to contain this new code

Reply:
Maybe. I chose not to be afraid of the extra indentation level, since much of the logic is shared. How about a `_get_geoip_urls()` helper?

Reply:
I tend to drop leading `get` from helpers like that but I don't feel strongly about it. I don't mind the extra indentation level, but collapsing 24 lines to _"get the relevant download URL"_ feels like a win for legibility to me here.

Reply:
Functions and methods should always have verb-y names, in my book, but if it helps: I didn't use `get`. 😁